app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx

```diff
@@ -148,7 +151,7 @@ function loadAmplifyConfig(
   return {
     ...baseConfig,
     ...platformConfig,
-    ...(urlOpener ? { urlOpener } : {}),
+    urlOpener,
```

Comment:
@somebody1234 this change was working before I merged the latest `develop` just now (so including changes from over the weekend). Now I get the following error when trying to build:

```shell
../dashboard/src/authentication/src/authentication/service.tsx:151:3 - error TS2375: Type '{ urlOpener: ((url: string) => void) | undefined; redirectSignIn: OAuthRedirect; redirectSignOut: OAuthRedirect; region: AwsRegion; scope: OAuthScope[]; responseType: OAuthResponseType; userPoolId: UserPoolId; userPoolWebClientId: UserPoolWebClientId; domain: OAuthDomain; } | { ...; }' is not assignable to type 'AmplifyConfig' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
  Type '{ urlOpener: ((url: string) => void) | undefined; redirectSignIn: auth.OAuthRedirect; redirectSignOut: auth.OAuthRedirect; region: AwsRegion; scope: OAuthScope[]; responseType: OAuthResponseType; userPoolId: auth.UserPoolId; userPoolWebClientId: auth.UserPoolWebClientId; domain: auth.OAuthDomain; }' is not assignable to type 'AmplifyConfig' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
    Types of property 'urlOpener' are incompatible.
      Type '((url: string) => void) | undefined' is not assignable to type 'OAuthUrlOpener'.
        Type 'undefined' is not assignable to type 'OAuthUrlOpener'.

151   return {
      ~~~~~~~~
152     ...baseConfig,
    ~~~~~~~~~~~~~~~~~~
... 
154     urlOpener,
    ~~~~~~~~~~~~~~
155   };
    ~~~~


Found 1 error in ../dashboard/src/authentication/src/authentication/service.tsx:151
```

Any idea what's happening here? `auth.AmplifyConfig` accepts `OAuthUrlOpener | undefined`, so this error doesn't make sense to me. It *should* be fine with `((url: string) => void) | undefined`

Reply:
After discussion on discord, this occurs because the `exactOptionalPropertyTypes` lint was enabled. This can be fixed by changing the type of the property from `urlOpener?: OAuthUrlOpener` to `urlOpener: OAuthUrlOpener | null`.